project/src/utils/JsonUtil.ts

```diff
@@ -181,27 +183,34 @@ export class JsonUtil {
     }
 
     /**
-     * Create file if nothing found
+     * Writes the file hashes to the cache path, to be used manually if writeHashes was set to false on deserializeWithCacheCheckAsync
+     */
+    public async writeCacheAsync(): Promise<void> {
+        await this.fileSystem.write(this.jsonCachePath, this.serialize(this.fileHashes, true));
+    }
+
+    /**
+     * Create file if nothing found asynchronously
      * @param jsonCachePath path to cache
      */
-    protected ensureJsonCacheExists(jsonCachePath: string): void {
+    protected async ensureJsonCacheExistsAsync(jsonCachePath: string): Promise<void> {
         if (!this.jsonCacheExists) {
-            if (!this.fileSystemSync.exists(jsonCachePath)) {
+            if (!(await this.fileSystem.exists(jsonCachePath))) {
                 // Create empty object at path
-                this.fileSystemSync.writeJson(jsonCachePath, {});
+                await this.fileSystem.writeJson(jsonCachePath, {});
             }
             this.jsonCacheExists = true;
         }
     }
 
     /**
-     * Read contents of json cache and add to class field
+     * Read contents of json cache and add to class field asynchronously
      * @param jsonCachePath Path to cache
      */
-    protected hydrateJsonCache(jsonCachePath: string): void {
+    protected async hydrateJsonCacheAsync(jsonCachePath: string): Promise<void> {
         // Get all file hashes
         if (!this.fileHashes) {
-            this.fileHashes = this.deserialize(this.fileSystemSync.read(`${jsonCachePath}`));
+            this.fileHashes = this.deserialize(await this.fileSystem.read(`${jsonCachePath}`));
```

Comment:
There's a `fileSystem.readJson` method that can be used to simplify here as well.